### ISSUE TITLE
Source controller does not preserve .imgpkg/images.yml as expected

### ISSUE THREAD

**Issue by mhoshi-vm** [vmware-tanzu/tanzu-source-controller#12]
**Bug description**  

Today I was trying to create a demo, involving image reallocation that needs `.imgpkg/images.yml` in place.
I start within in image bundle like the following. *The important part is that it is pointing to image:harbor.run01*

```
% imgpkg pull -b harbor.run01.lespaulstudioplus.info/prod-deploy/py-func -o /tmp/bundle > /dev/null && cat /tmp/bundle/.imgpkg/images.yml
---
apiVersion: imgpkg.carvel.dev/v1alpha1
images:
- annotations:
    kbld.carvel.dev/id: harbor.lespaulstudioplus.info/tap/supply-chain/py-func-python-demo@sha256:506ee0eb7bb12e77ad01d9fc9e183470826431697a00065d6e41b83b95c0ba1f
  image: harbor.run01.lespaulstudioplus.info/prod-deploy/py-func@sha256:506ee0eb7bb12e77ad01d9fc9e183470826431697a00065d6e41b83b95c0ba1f
kind: ImagesLock
```

Now I will upload this image bundle to the run cluster, and create an deliverable
The source-controller-manager will create an imagerepository resource

```
NAME               IMAGE                                                     URL                                                                                                                                                                                                   READY   REASON   AGE
py-func-delivery   harbor.run01.lespaulstudioplus.info/prod-deploy/py-func   http://source-controller-manager-artifact-service.source-system.svc.cluster.local/./imagerepository/dev-ns1/py-func-delivery/d62f9c1525ae72791517ebfc0e788075ee218d1ef6512f7b10fc4ca8ea7dfab6.tar.gz   True    Ready    16s
```

All expected nothing wrong.
From here, I found the unexpected. I was finding myself having trouble for the image reallocation to work, until i realized the archive file that the source-controller created has a wrong .imgpkg/images.yml *The image: is no longer pointing to harbor.run01 instead the original image*

```
root@hoge:/# curl http://source-controller-manager-artifact-service.source-system.svc.cluster.local/./imagerepository/dev-ns1/py-func-delivery/d62f9c1525ae72791517ebfc0e788075ee218d1ef6512f7b10fc4ca8ea7dfab6.tar.gz --output source.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1279  100  1279    0     0   178k      0 --:--:-- --:--:-- --:--:--  178k
root@hoge:/# tar xvf source.tar.gz
.imgpkg/images.yml
delivery.yml
files.json
root@hoge:/# cat .imgpkg/images.yml
---
apiVersion: imgpkg.carvel.dev/v1alpha1
images:
- annotations:
    kbld.carvel.dev/id: harbor.lespaulstudioplus.info/tap/supply-chain/py-func-python-demo@sha256:506ee0eb7bb12e77ad01d9fc9e183470826431697a00065d6e41b83b95c0ba1f
  image: harbor.lespaulstudioplus.info/tap/supply-chain/py-func-python-demo@sha256:506ee0eb7bb12e77ad01d9fc9e183470826431697a00065d6e41b83b95c0ba1f
kind: ImagesLock
```

I know we don't support this scenario, but we are getting a lot of request to demo TAP in air-gap env, using rellocation from build clusters.
So to summarize, is this known limitation ?

**Expected behavior**

imgpkg lock file preserved during bundle upload

**Steps to reproduce the bug**

upload a bundle and create a deliverable based on it

**Version** (include the SHA if the version is not obvious)

On my laptop i use imgpkg 0.33.0 
the source controller used is from TAP 1.4

**Environment where the bug was observed (cloud, OS, etc)**

vSphere

**Relevant Debug Output (Logs, etc)**
